src/test/kotlin/logic/usecase/GetMealsByAddedDateUseCaseTest.kt

```
@@ -0,0 +1,104 @@
+package logic.usecase
+
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.datetime.LocalDate
+import logic.MealsRepository
+import logic.exception.FoodMoodException
+import logic.helpers.createMeal
+import logic.models.Meal
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.assertThrows
+import kotlin.test.Test
+
+class GetMealsByAddedDateUseCaseTest {
+    lateinit var getMealsByAddedDateUseCase: GetMealsByAddedDateUseCase
+    lateinit var mealsRepository: MealsRepository
+    val meals = listOf<Meal>(
+        createMeal(submittedDate = LocalDate(2025, 9, 16), id = "50"),
+        createMeal(submittedDate = LocalDate(2005, 9, 16), id = "120"),
+        createMeal(submittedDate = LocalDate(2005, 9, 16), id = "13"),
+    )
+
+    @BeforeEach
+    fun setup() {
+        mealsRepository = mockk(relaxed = true)
+        getMealsByAddedDateUseCase = GetMealsByAddedDateUseCase(mealsRepository)
+    }
+
+    @Test
+    fun `getMealsByDate should throw InvalidDateFormatException when it given invalid date format`() {
```

also  make the begin of function name is should keyword like that:
` should getMealsByDate throw InvalidDateFormatException when it given invalid date format`

---

i guess this is like a question should end with ? -> should getMealsByDate throw InvalidDateFormatException when it given invalid date format ?

---

remember that the name of function can not start with name  ,

---

Done